monitor.cpp

```diff
@@ -3,36 +3,139 @@
 #include <thread>
 #include <chrono>
 #include <iostream>
-using std::cout, std::flush, std::this_thread::sleep_for, std::chrono::seconds;
+#include <string>
+#include <vector>
 
-int vitalsOk(float temperature, float pulseRate, float spo2) {
-  if (temperature > 102 || temperature < 95) {
-    cout << "Temperature is critical!\n";
-    for (int i = 0; i < 6; i++) {
-      cout << "\r* " << flush;
-      sleep_for(seconds(1));
-      cout << "\r *" << flush;
-      sleep_for(seconds(1));
-    }
-    return 0;
-  } else if (pulseRate < 60 || pulseRate > 100) {
-    cout << "Pulse Rate is out of range!\n";
-    for (int i = 0; i < 6; i++) {
-      cout << "\r* " << flush;
-      sleep_for(seconds(1));
-      cout << "\r *" << flush;
-      sleep_for(seconds(1));
-    }
-    return 0;
-  } else if (spo2 < 90) {
-    cout << "Oxygen Saturation out of range!\n";
+CheckVitals::Language currentLanguage = CheckVitals::Language::English; // Default language
+static std::string transMessage = "";
+
+//CheckVitals::~CheckVitals()
+//{
+//     delete message;
+//}
+
+void CheckVitals::displayWarning(const std::string& message) {
+    std::cout << message << "\n";
     for (int i = 0; i < 6; i++) {
-      cout << "\r* " << flush;
-      sleep_for(seconds(1));
-      cout << "\r *" << flush;
-      sleep_for(seconds(1));
-    }
-    return 0;
-  }
-  return 1;
+        std::cout << "\r* " << std::flush;
+        std::this_thread::sleep_for(std::chrono::seconds(1));
+        std::cout << "\r *" << std::flush;
+        std::this_thread::sleep_for(std::chrono::seconds(1));
+    }
+}
+
+bool isVitalInRange(const double min, const double max, const double value) {
+    return value >= min && value <= max;
+}
+
+// Function to retrieve the translation for the given key
+std::string getTranslation(const std::string& key) {
+    std::string str = "";
+    if (translations.find(key) != translations.end()) {
+        const Translation& translation = translations[key];
+        
+        switch (currentLanguage){
+        case CheckVitals::Language::English:
+            str = translation.english;
+            break;
+        case CheckVitals::Language::German:
+            str = translation.german;
+            break;
+        }
+    }
+    return str;
+}
+
+// Helper functions to check each vital
+bool isTemperatureNotNormal(double temperature) {
```

Comment:
This is exceeding the complexity limit?
Try splitting the function. One way is to take the "positive" route and say `isTemperatureNormal`. Return if temperature is normal, otherwise call another function to judge between warning and critical-breach.
Like this, keep splitting the functions to see where we end up.